cypress/e2e/supervision/clients/documents/open-document.cy.js

```diff
@@ -0,0 +1,67 @@
+import path from "path";
+
+beforeEach(() => {
+  cy.loginAs("Allocations User");
+  cy.createAClient();
+  cy.loginAs("Public API");
+  cy.uploadDocument();
+});
+
+describe(
+  "Open document successfully",
+  { tags: ["@supervision-core", "@documents", "@open-document", "@smoke-journey"] },
+  () => {
+    it("opens a document successfully", () => {
+      cy.loginAs("Case Manager");
+      cy.get("@clientId").then((clientId) => {
+        cy.visit(`/supervision/#/clients/${clientId}`);
+        cy.contains("Ted Tedson");
+      });
+
+      cy.get(".TABS_DOCUMENTS").click();
+      cy.get(".filter-numbers > .number").should("have.text", 1);
+      cy.get("#select-all-documents-checkbox").check({ force: true });
+
+      cy.task('listContentsOfDownloadsFolder', Cypress.config("downloadsFolder")).then(beforeDownloadList => {
+        cy.contains(".button", "Open").click().wait(1000);
+        cy.task('listContentsOfDownloadsFolder', Cypress.config("downloadsFolder")).then(afterDownloadList => {
+          const newFilename = afterDownloadList.filter(file => !beforeDownloadList.includes(file))[0]
+          const newFilePath = path.join(
+            Cypress.config("downloadsFolder"),
+            newFilename
+          );
+          cy.verifyDownload(newFilename, {contains: true});
+          cy.readFile(newFilePath)
+        })
+      })
+    })
+
+    it("multiple document download", () => {
+      cy.uploadDocument()
+      cy.loginAs("Case Manager");
+      cy.get("@clientId").then((clientId) => {
+        cy.visit(`/supervision/#/clients/${clientId}`);
+        cy.contains("Ted Tedson");
+      });
+
+      cy.task('listContentsOfDownloadsFolder', Cypress.config("downloadsFolder")).then(() => {
+        cy.get(".TABS_DOCUMENTS").click();
```

Comment:
I think I created confusion by talking about the then statements when there were multiple. 

Rather than moving the tab click inside the download config, I meant moving the tab click after the `cy.contains("Ted Tedson");` otherwise cypress might try and click the tab before its created a client and moved to the page of the client. The way cypress does retries might mask this slightly.

Reply:
Lol, I didn't even notice that then.